src/WebJobs.Script/Rpc/Configuration/WorkerConfigFactory.cs

```diff
@@ -220,5 +225,27 @@ internal string GetExecutablePathForJava(string defaultExecutablePath)
                 return Path.GetFullPath(Path.Combine(javaHome, "bin", defaultExecutablePath));
             }
         }
+
+        internal string GetHydratedWorkerPath(WorkerDescription description)
+        {
+            string workerPath = description.GetWorkerPath();
+            if (string.IsNullOrEmpty(workerPath))
+            {
+                return null;
+            }
+
+            string os = _systemRuntimeInformation.GetOSPlatform().ToString();
+
+            string architecture = _systemRuntimeInformation.GetOSArchitecture().ToString();
+            string version = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeVersionSettingName);
+            if (string.IsNullOrEmpty(version))
+            {
+                version = description.DefaultLanguageVersion;
+            }
+
+            return workerPath.Replace("{os}", os)
```

Comment:
As discussed offline, if the worker opts into the new structure, add validation on runtime version, os and architechture